scripts/jenkins/ardana/ansible/roles/crowbar_register/tasks/reserve_admin_ip.yml

```diff
@@ -0,0 +1,43 @@
+---
+
+- name: Get crowbar admin_network databag contents
+  shell: |
+    knife data bag show crowbar admin_network -F json
+  delegate_to: "{{ ardana_env }}"
+  register: _register_databag
+  run_once: true
+
+- name: Set databag contents as fact
+  set_fact:
+    admin_network_databag: "{{ _register_databag.stdout | from_json }}"
+  run_once: true
+
+- name: Set databag note entry fact
+  set_fact:
+    admin_network_databag_entry:
+      machine: "{{ crowbar_node_name }}"
+      address: "{{ ansible_host }}"
+      interface: "intf{{ admin_interface_index }}"
+
+- name: Update admin network databag allocations
+  set_fact:
+    admin_network_databag: >
+      {%- set _ret = admin_network_databag -%}
+      {%- set _entry = { hostvars[item]['crowbar_node_name']: hostvars[item]['admin_network_databag_entry'] } -%}
+      {%- set _ = admin_network_databag['allocated_by_name'].update( _entry ) -%}
+      {{ _ret }}
+  loop: "{{ groups['cloud_virt_hosts'] }}"
+  run_once: true
+
+- name: Save updated databag contents
+  copy:
+    content: "{{ admin_network_databag | to_json }}"
+    dest: "/tmp/admin_network.json"
```

Comment:
Could this task be executed from multiple nodes at the same time? If so, then this file could be overwritten by others.

Reply:
That's where the `run_once: true` part comes in: it forces this task to be executed only once.